### PR TITLE
Revise comments in ExpressionExtensions to fix code analysis warnings

### PR DESCRIPTION
**Pull request by TimLovellSmith**

### CODE REVIEW

File: src/FakeItEasy/ExpressionExtensions.cs [hunk 1]
```diff
@@ -14,35 +14,29 @@ internal static class ExpressionExtensions
...
             return EvaluateOptimized(expression);
         }
-        
+
+        // About the optimizations: 
+        // Key observation is that walking the expression tree and evaluating directly can be much faster for simple expressions like:
+        // - constant expressions (null, 1, "easy", etc.)
+        // - local variables referenced from lambda expressions, such as 'fake' in the A.CallTo line:
+        //     {
+        //         var fake = A.Fake&lt;Something&gt;();
```
Comment (thomaslevesque) [inline src/FakeItEasy/ExpressionExtensions.cs]: This is no longer an XML doc comment, so you don't need to encode the `<` and `>`

**Comment by thomaslevesque** [comment]
Looking good, thanks @TimLovellSmith!

> @FakeItEasy/owners, how do we feel about merging this and then the remainder of @thomaslevesque's changes?

:+1: 

There will be a conflict, but it'll be easy to fix

File: src/FakeItEasy/ExpressionExtensions.cs (hunk 1 above)
Comment (blairconrad) [inline src/FakeItEasy/ExpressionExtensions.cs]: Bah. Good eye.

**Comment by blairconrad** [comment]
Thanks for merging, @thomaslevesque. I figured you'd be up to fixing the conflict!

File: src/FakeItEasy/ExpressionExtensions.cs (hunk 1 above)
Comment (thomaslevesque) [inline src/FakeItEasy/ExpressionExtensions.cs]: Would have been more useful if I had waited for the fix before merging... don't know what I was thinking :confused:

Comment (blairconrad) [inline src/FakeItEasy/ExpressionExtensions.cs]: I'd not even noticed the escaped chars at the time.
Although when I first saw the PR, I saw this:
image

and thought they'd been unescaped. Silly of me.

Comment (thomaslevesque) [inline src/FakeItEasy/ExpressionExtensions.cs]: Ah, you switched to a font with ligatures ? Me too, and I love it :)

**Pull request merged**